rust-bindings/src/lib.rs

```diff
@@ -182,45 +184,90 @@ pub fn get_receive_contract_parameter_aux(
     contract_name: &str,
     entrypoint: &str,
     value: &[u8],
-) -> Result<Vec<u8>> {
+) -> Result<Vec<u8>, FFIError> {
     let module_schema = VersionedModuleSchema::new(schema, &schema_version)?;
     let parameter_type = module_schema.get_receive_param_schema(contract_name, entrypoint)?;
-    let deserialized = deserialize_type_value(value, &parameter_type, true)?;
+    let deserialized = deserialize_type_value(value, &parameter_type)?;
     Ok(deserialized)
 }
 
-fn schema_display_aux(schema: &[u8], schema_version: Option<u8>) -> Result<Vec<u8>> {
+fn schema_display_aux(schema: &[u8], schema_version: Option<u8>) -> Result<Vec<u8>, FFIError> {
     let display = VersionedModuleSchema::new(schema, &schema_version)?;
     Ok(display.to_string().into_bytes())
 }
 
+#[derive(Error, Debug)]
+pub enum FFIError {
+    #[error("{0}")]
+    JsonError(String),
+    #[error("error when using serde")]
+    SerdeJsonError,
+    #[error("encountered string which wasn't utf8 encoded")]
+    Utf8Error,
+    #[error(transparent)]
+    VersionedSchemaError(#[from] VersionedSchemaError),
+}
+
+impl FFIError {
+    fn to_int(&self) -> u16 {
```

Comment:
**Suggestion:** It would be good with a comment here to why it is not starting from `0`